feat(core)!: WIP make tenant id required

nkwiatkowski opened the pull request:
<!-- Thanks for submitting a Pull Request to Kestra. To help us review your contribution, please follow the guidelines below:

- Make sure that your commits follow the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification e.g. `feat(ui): add a new navigation menu item` or `fix(core): fix a bug in the core model` or `docs: update the README.md`. This will help us automatically generate the changelog.
- The title should briefly summarize the proposed changes.
- Provide a short overview of the change and the value it adds.
- Share a flow example to help the reviewer understand and QA the change.
- Use "closes" to automatically close an issue. For example, `closes #1234` will close issue #1234. -->
closes #https://github.com/kestra-io/kestra-ee/issues/3427
### What changes are being made and why?

<!-- Please include a brief summary of the changes included in this PR e.g. closes #1234. -->

---

### How the changes have been QAed?

<!-- Include example code that shows how this PR has been QAed. The code should present a complete yet easily reproducible flow.

```yaml
# Your example flow code here
```

Note that this is not a replacement for unit tests but rather a way to demonstrate how the changes work in a real-life scenario, as the end-user would experience them.

Remove this section if this change applies to all flows or to the documentation only. -->

---

### Setup Instructions

<!--If there are any setup requirements like API keys or trial accounts, kindly include brief bullet-points-description outlining the setup process below.

- [External System Documentation](URL)
- Steps to set up the necessary resources

If there are no setup requirements, you can remove this section.

Thank you for your contribution. ❤️  -->

loicmathieu committed:
feat(plugin): add a way to provide additional type of plugins

Provide a way for plugins to define a new type of plugins.
To do that, a plugin must provide both an abstract base class that extends AdditionalPlugin and a set of concret classes.
Both the abstract base class and the concrete classes mut be inside the same plugin. This is a limitation that we may work on later by providing, for example, an SPI to add base classes to the application classloader.

loicmathieu committed:
feat(system): add TestSuite model,taskFixture impl

loicmathieu committed:
fix(execution)*: decode and hide nested inputs of type SECRET

Fixes #7964

loicmathieu committed:
feat(executions): Add workerId to each worker task attemps

Closes #7799

AcevedoR committed:
feat(core): forward execution labels in Flow Trigger

AcevedoR committed:
tests(webserver): fix flaky test which could query previous tests tasks

AcevedoR committed:
tests(webserver): fix flaky test which could query previous tests tasks

fhussonnois committed:
fix(system): change default config values for liveness

Change kestra.server.liveness.interval from 5s to 10s
to be less agressive on liveness check. Align
other default liveness configs with kafka implementation.

dependabot[bot] committed:
build(deps): bump com.github.oshi:oshi-core from 6.8.0 to 6.8.1

Bumps [com.github.oshi:oshi-core](https://github.com/oshi/oshi) from 6.8.0 to 6.8.1.
- [Release notes](https://github.com/oshi/oshi/releases)
- [Changelog](https://github.com/oshi/oshi/blob/master/CHANGELOG.md)
- [Commits](https://github.com/oshi/oshi/compare/oshi-parent-6.8.0...oshi-parent-6.8.1)

---
updated-dependencies:
- dependency-name: com.github.oshi:oshi-core
  dependency-version: 6.8.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump software.amazon.awssdk:bom from 2.31.21 to 2.31.25

Bumps software.amazon.awssdk:bom from 2.31.21 to 2.31.25.

---
updated-dependencies:
- dependency-name: software.amazon.awssdk:bom
  dependency-version: 2.31.25
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump flyingSaucerVersion from 9.11.6 to 9.12.0

Bumps `flyingSaucerVersion` from 9.11.6 to 9.12.0.

Updates `org.xhtmlrenderer:flying-saucer-core` from 9.11.6 to 9.12.0
- [Release notes](https://github.com/flyingsaucerproject/flyingsaucer/releases)
- [Changelog](https://github.com/flyingsaucerproject/flyingsaucer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/flyingsaucerproject/flyingsaucer/compare/v9.11.6...v9.12.0)

Updates `org.xhtmlrenderer:flying-saucer-pdf` from 9.11.6 to 9.12.0
- [Release notes](https://github.com/flyingsaucerproject/flyingsaucer/releases)
- [Changelog](https://github.com/flyingsaucerproject/flyingsaucer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/flyingsaucerproject/flyingsaucer/compare/v9.11.6...v9.12.0)

---
updated-dependencies:
- dependency-name: org.xhtmlrenderer:flying-saucer-core
  dependency-version: 9.12.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: org.xhtmlrenderer:flying-saucer-pdf
  dependency-version: 9.12.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump io.micronaut.platform:micronaut-platform

Bumps [io.micronaut.platform:micronaut-platform](https://github.com/micronaut-projects/micronaut-platform) from 4.8.0 to 4.8.2.
- [Release notes](https://github.com/micronaut-projects/micronaut-platform/releases)
- [Commits](https://github.com/micronaut-projects/micronaut-platform/compare/v4.8.0...v4.8.2)

---
updated-dependencies:
- dependency-name: io.micronaut.platform:micronaut-platform
  dependency-version: 4.8.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump co.elastic.logging:logback-ecs-encoder

Bumps [co.elastic.logging:logback-ecs-encoder](https://github.com/elastic/ecs-logging-java) from 1.6.0 to 1.7.0.
- [Release notes](https://github.com/elastic/ecs-logging-java/releases)
- [Commits](https://github.com/elastic/ecs-logging-java/compare/v1.6.0...v1.7.0)

---
updated-dependencies:
- dependency-name: co.elastic.logging:logback-ecs-encoder
  dependency-version: 1.7.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump com.google.guava:guava from 33.4.7-jre to 33.4.8-jre

Bumps [com.google.guava:guava](https://github.com/google/guava) from 33.4.7-jre to 33.4.8-jre.
- [Release notes](https://github.com/google/guava/releases)
- [Commits](https://github.com/google/guava/commits)

---
updated-dependencies:
- dependency-name: com.google.guava:guava
  dependency-version: 33.4.8-jre
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump org.opensearch.client:opensearch-java

Bumps [org.opensearch.client:opensearch-java](https://github.com/opensearch-project/opensearch-java) from 2.22.0 to 2.23.0.
- [Release notes](https://github.com/opensearch-project/opensearch-java/releases)
- [Changelog](https://github.com/opensearch-project/opensearch-java/blob/v2.23.0/CHANGELOG.md)
- [Commits](https://github.com/opensearch-project/opensearch-java/compare/v2.22.0...v2.23.0)

---
updated-dependencies:
- dependency-name: org.opensearch.client:opensearch-java
  dependency-version: 2.23.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
feat(core)!: WIP make tenant id required

Commit:
feat(core)!: WIP make tenant id required

Commit:
feat(core)!: WIP make tenant id required

Commit:
feat(core)!: WIP make tenant id required

Commit:
Merge remote-tracking branch 'origin/feat/add_main_tenant' into feat/add_main_tenant

# Conflicts:
#	core/src/test/java/io/kestra/core/repositories/AbstractExecutionRepositoryTest.java
#	core/src/test/java/io/kestra/core/repositories/AbstractFlowRepositoryTest.java
#	core/src/test/java/io/kestra/core/runners/NoEncryptionConfiguredTest.java
#	webserver/src/test/java/io/kestra/webserver/controllers/api/ExecutionControllerRunnerTest.java
#	webserver/src/test/java/io/kestra/webserver/controllers/api/ExecutionControllerTest.java
#	webserver/src/test/java/io/kestra/webserver/controllers/api/KVControllerTest.java
#	webserver/src/test/java/io/kestra/webserver/controllers/api/NamespaceFileControllerTest.java
#	webserver/src/test/java/io/kestra/webserver/controllers/api/TemplateControllerTest.java

loicmathieu committed:
feat(flows): Allow to define an onPause task on the Pause task

The onPause task will be executed immediatly when the execution is paused.
Part-of: #3601

Commit:
test(core)!: WIP fix storage unit test

AcevedoR committed:
tests(system): isolate SchedulerScheduleTest tests with tenantId

AcevedoR committed:
tests(system): debug flaky error of shouldPauseExecutionByQueryRunningFlows

loicmathieu committed:
Feat/storage outputs (#8361)

* feat(executions): Store outputs inside the internal storage (1/2)

* feat(executions): Store outputs inside the internal storage (2/2)

* feat(test): allow passing tenantId to tests

---------

Co-authored-by: Ludovic DEHON <tchiot.ludo@gmail.com>

loicmathieu committed:
Feat/storage outputs (#8361)

* feat(executions): Store outputs inside the internal storage (1/2)

* feat(executions): Store outputs inside the internal storage (2/2)

* feat(test): allow passing tenantId to tests

---------

Co-authored-by: Ludovic DEHON <tchiot.ludo@gmail.com>

loicmathieu committed:
Feat/storage outputs (#8361)

* feat(executions): Store outputs inside the internal storage (1/2)

* feat(executions): Store outputs inside the internal storage (2/2)

* feat(test): allow passing tenantId to tests

---------

Co-authored-by: Ludovic DEHON <tchiot.ludo@gmail.com>

loicmathieu committed:
fix(executions): unqueing execution must remove the execution queued

When an execution is queued in the JDBC backend, a record is inserted inside the execution_queued table, we must remove this record when we unqeue an execution.

Fixes #8448

AcevedoR committed:
tests(system): debug flaky error of shouldPauseExecutionByQueryRunningFlows

AcevedoR committed:
tests(system): try with different task id

AcevedoR committed:
tests(system): bump sleep-short sleep time to 10s

MilosPaunovic committed:
chore(deps): regular dependency update (#8484)

Performing a weekly round of dependency updates in the NPM ecosystem to keep everything up to date.

Piyush-r-bhaskar committed:
fix(ui): full view height for single task logs (#8042)

Co-authored-by: Miloš Paunović <paun992@hotmail.com>

elevatebart committed:
feat: synchronize task edition with editor (#8433)

Piyush-r-bhaskar committed:
fix(controls): adjust bottom position of contorls in multiPanelsEditor (#8465)

github-actions[bot] committed:
chore(core): localize to languages other than english (#8485)

Extended localization support by adding translations for multiple languages using English as the base. This enhances accessibility and usability for non-English-speaking users while keeping English as the source reference.

Co-authored-by: GitHub Action <actions@github.com>

MilosPaunovic committed:
refactor(core): pass the dynamic concurrency schema to no code editor (#8488)

There was an issue with passing hard-coded concurrency schema to be rendered in No Code editor, which is now amended and we're passing down the previously fetched one

tchiotludo committed:
chore(deps): update gradle version

aj-emerich committed:
doc(basic.md): add link to configuration for kestra property variables (#8490)

Piyush-r-bhaskar committed:
fix(flows): properly check average duration for dashboard graphs (#8457)

There was a problem on flows view with the main chart not showing proper data until user clicks on duration toggle.

Closes https://github.com/kestra-io/kestra/issues/8435.
Closes https://github.com/kestra-io/kestra-ee/issues/3499.

aj-emerich committed:
docs(core-pause): update pauseDuration properties, titles, descriptions (#8495)

loicmathieu committed:
fix(system): restrict the JdbcConcurrencyLimitService to the JDBC runner

aj-emerich committed:
fix(core): fix indexer metric description (#8500)

anna-geller commented:
@nkwiatkowski perhaps worth marking the PR in draft status? (since WIP)

Ben8t committed:
Add examples with expression and trimmed values (#6154)

Piyush-r-bhaskar committed:
chore(ui): improvement to drilldown for Default and Custom Charts. (#7885)

* chore(ui): improvement to drilldown for Default and Custom Charts.

* minor tweak

* test: fix the Barchart stories to test drilldown

---------

Co-authored-by: Bart Ledoux <bledoux@kestra.io>

satvik2131 committed:
fix(ui): restart trigger position for backfill column (#8246)

Co-authored-by: Miloš Paunović <paun992@hotmail.com>
Co-authored-by: Barthélémy Ledoux <ledouxb@me.com>
Co-authored-by: Bart Ledoux <bledoux@kestra.io>

lw-yang committed:
fix(jdbc): add service_id index on service_instance table

loicmathieu committed:
fix(core): failing DocumentationGeneratorTest.returnDoc()

loicmathieu committed:
feat(system)!: remove the SQLServer runner

Part-of: https://github.com/kestra-io/kestra-ee/issues/3504

loicmathieu committed:
chore(build): add Postgres stat extension

Commit:
fix(ui): save existing flow after making changes (#8378)

Co-authored-by: Barthélémy Ledoux <ledouxb@me.com>

Piyush-r-bhaskar committed:
chore(core): refactor  component to composition API  structure and with some styling (#8504)

hashimzs committed:
feat(flows): add validation for use of inputs and outputs with '-' in the name (#8379)

nkwiatkowski committed:
test(core): fix breaking change in local flow repository (#8517)

Co-authored-by: nKwiatkowski <nkwiatkowski@kestra.io>

loicmathieu committed:
feat(plugins): add Langchain4J plugins

lw-yang committed:
chore(system): add warn log when emit logQueue failed (#8432)

loicmathieu committed:
feat(plugins): add Go Script plugin

Piyush-r-bhaskar committed:
chore(flows): improve the blueprints view within the flow editing panels (#7983)

Changes here consist of removing the tags from blueprint view on Multi Panel flow editor, along with couple of other UI improvements.

Closes https://github.com/kestra-io/kestra/issues/7881.

Co-authored-by: Miloš Paunović <paun992@hotmail.com>

MilosPaunovic committed:
fix(flows): properly load blueprints in multi panel view (#8524)

While using the new Multi Panel view blueprints were not loading properly due to wrong paramtere being sent to action. now that's sorted.

Closes https://github.com/kestra-io/kestra/issues/8523.

Piyush-r-bhaskar committed:
feat(flows): improve the display of array inputs when running an execution (#7953)

This PR is introducing a change of how the `array` inputs are displayed inside the flow run dialog, to be more user-friendly.

Closes https://github.com/kestra-io/kestra/issues/6947.

---------

Co-authored-by: Miloš Paunović <paun992@hotmail.com>

aj-emerich committed:
docs(flow-trigger): add note about no Pebble in conditions

yuri1969 committed:
fix(ui): amend Absolute date filter's looks (#8501)

github-actions[bot] committed:
chore(core): localize to languages other than english (#8528)

Extended localization support by adding translations for multiple languages using English as the base. This enhances accessibility and usability for non-English-speaking users while keeping English as the source reference.

Co-authored-by: GitHub Action <actions@github.com>

loicmathieu committed:
fix(triggers): inject default later inside the Scheduler

Today, as they are injected eagerly, they are done even if no trigger exists.
This is counter-performant, and in case the flow is an error will log each seconds.
Doing it a little later will be better.

LeNeRoTeX commented on core/src/main/java/io/kestra/core/tenant/TenantService.java:
```diff
@@ -5,13 +5,15 @@
+
     /**
      * Resolve the current tenant and return its identifier.
      * If the tenant is the default tenant, it returns null, which is always the case on OSS as Tenant is an EE feature.
```
"If the tenant is the default tenant, it returns 'main', which is always the case on OSS as Tenant is an EE feature.
Comment"

nkwiatkowski replied:
Yes, but tenant id will still be required, so it will always return "main" in OSS to keep the code similar to EE

LeNeRoTeX replied:
Yes, i mean the doc above need a change from "it returns null" to "it returns 'main'"

rajatsingh23 committed:
fix(ui): remove parts of filter using backspace (#8105)

Co-authored-by: Miloš Paunović <paun992@hotmail.com>
Co-authored-by: Barthélémy Ledoux <ledouxb@me.com>

Piyush-r-bhaskar committed:
fix(ui): open link in markdown in other tab. (#8258)

* fix(ui): open link in markdown in other tab.

* chore(core): restrict attribute to external links.

Co-authored-by: Barthélémy Ledoux <ledouxb@me.com>

---------

Co-authored-by: Barthélémy Ledoux <ledouxb@me.com>

loicmathieu committed:
fix(system): load OpenTelemetry lib in the app classloader

Without that, as we have it here, plugins may have class loading issue if they use OpenTelemetry internally (like in the Elasticsearch client).

dependabot[bot] committed:
build(deps): bump software.amazon.awssdk:bom from 2.31.25 to 2.31.30

Bumps software.amazon.awssdk:bom from 2.31.25 to 2.31.30.

---
updated-dependencies:
- dependency-name: software.amazon.awssdk:bom
  dependency-version: 2.31.30
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump jacksonVersion from 2.18.3 to 2.19.0

Bumps `jacksonVersion` from 2.18.3 to 2.19.0.

Updates `com.fasterxml.jackson:jackson-bom` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson-bom/compare/jackson-bom-2.18.3...jackson-bom-2.19.0)

Updates `com.fasterxml.jackson.core:jackson-core` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson-core/compare/jackson-core-2.18.3...jackson-core-2.19.0)

Updates `com.fasterxml.jackson.core:jackson-databind` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson/commits)

Updates `com.fasterxml.jackson.core:jackson-annotations` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson/commits)

Updates `com.fasterxml.jackson.module:jackson-module-parameter-names` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson-modules-java8/compare/jackson-modules-java8-2.18.3...jackson-modules-java8-2.19.0)

Updates `com.fasterxml.jackson.dataformat:jackson-dataformat-yaml` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson-dataformats-text/compare/jackson-dataformats-text-2.18.3...jackson-dataformats-text-2.19.0)

Updates `com.fasterxml.jackson.dataformat:jackson-dataformat-smile` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson-dataformats-binary/compare/jackson-dataformats-binary-2.18.3...jackson-dataformats-binary-2.19.0)

Updates `com.fasterxml.jackson.dataformat:jackson-dataformat-cbor` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson-dataformats-binary/compare/jackson-dataformats-binary-2.18.3...jackson-dataformats-binary-2.19.0)

Updates `com.fasterxml.jackson.dataformat:jackson-dataformat-ion` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson-dataformat-ion/commits)

Updates `com.fasterxml.jackson.dataformat:jackson-dataformat-xml` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson-dataformat-xml/compare/jackson-dataformat-xml-2.18.3...jackson-dataformat-xml-2.19.0)

Updates `com.fasterxml.jackson.datatype:jackson-datatype-guava` from 2.18.3 to 2.19.0
- [Commits](https://github.com/FasterXML/jackson-datatypes-collections/compare/jackson-datatypes-collections-2.18.3...jackson-datatypes-collections-2.19.0)

Updates `com.fasterxml.jackson.datatype:jackson-datatype-jsr310` from 2.18.3 to 2.19.0

Updates `com.fasterxml.jackson.datatype:jackson-datatype-jdk8` from 2.18.3 to 2.19.0

---
updated-dependencies:
- dependency-name: com.fasterxml.jackson:jackson-bom
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.core:jackson-core
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.core:jackson-databind
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.core:jackson-annotations
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.module:jackson-module-parameter-names
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.dataformat:jackson-dataformat-yaml
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.dataformat:jackson-dataformat-smile
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.dataformat:jackson-dataformat-cbor
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.dataformat:jackson-dataformat-ion
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.dataformat:jackson-dataformat-xml
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.datatype:jackson-datatype-guava
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.datatype:jackson-datatype-jsr310
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
- dependency-name: com.fasterxml.jackson.datatype:jackson-datatype-jdk8
  dependency-version: 2.19.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): bump org.wiremock:wiremock-jetty12 from 3.12.1 to 3.13.0

Bumps [org.wiremock:wiremock-jetty12](https://github.com/wiremock/wiremock) from 3.12.1 to 3.13.0.
- [Release notes](https://github.com/wiremock/wiremock/releases)
- [Commits](https://github.com/wiremock/wiremock/compare/3.12.1...3.13.0)

---
updated-dependencies:
- dependency-name: org.wiremock:wiremock-jetty12
  dependency-version: 3.13.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
feat(ui): Add search in internal docs (#8458)

Co-authored-by: Barthélémy Ledoux <ledouxb@me.com>

Piyush-r-bhaskar committed:
chore(execution): update display names for executions. (#8527)

nkwiatkowski committed:
fix(core): change incorrectly used search parameter (#8534)

Co-authored-by: nKwiatkowski <nkwiatkowski@kestra.io>

elevatebart committed:
fix(ui): set isCreating to false when opening flow edit mode (#8549)

Piyush-r-bhaskar committed:
fix(core): safely access section and identifier query params (#8542)

Co-authored-by: Barthélémy Ledoux <ledouxb@me.com>

elevatebart committed:
fix(ui): update storybook editor tests with provided keys (#8550)

Co-authored-by: Piyush Bhaskar <102078527+Piyush-r-bhaskar@users.noreply.github.com>

github-actions[bot] committed:
chore(core): localize to languages other than english (#8554)

Extended localization support by adding translations for multiple languages using English as the base. This enhances accessibility and usability for non-English-speaking users while keeping English as the source reference.

Co-authored-by: GitHub Action <actions@github.com>

MilosPaunovic committed:
fix(triggers): amend broken filtering on triggers tab (#8553)

There was a problem with both namespace and state filters on Triggers page which is now properly sorted.

Closes https://github.com/kestra-io/kestra/issues/8529.

yuri1969 committed:
chore: attempt to fix flaky tests (#8537)

SingleFlowCommandsTest:

The flow Delete -> Create -> Update sequence is weird - delete got HTTP 404.
Reworked to Create -> Update -> Delete sequence.

PurgeLogsTest:

The log repository contained the prepared single entry but also might
contain additional entries from previously logged messages.

nkwiatkowski committed:
fix(namespaces): namespaceFiles with same name are wrongly overwritten (#8562)

Co-authored-by: nKwiatkowski <nkwiatkowski@kestra.io>

fhussonnois committed:
fix(executions): fix execution failure due to UnsupportedOperationException (#8563)

Fix: #8563

Piyush-r-bhaskar committed:
chore(triggers)*: properly handle switches for triggers disabled from within flow source (#8106)

It was not clear as to which trigger can not be enabled and why. Now, that is much more clear with the proper tooltips and disabling of switch toggling.

Closes https://github.com/kestra-io/kestra/issues/8011.
Closes https://github.com/kestra-io/kestra/issues/5736.

github-actions[bot] committed:
chore(core): localize to languages other than english (#8568)

Extended localization support by adding translations for multiple languages using English as the base. This enhances accessibility and usability for non-English-speaking users while keeping English as the source reference.

Co-authored-by: GitHub Action <actions@github.com>

loicmathieu committed:
feat(system)*: decrease defaut JDBC queue poll size

Decreasing it from 100 to 50 didn't show any performance hit but should lower the memory consumption now that we process the queue concurrently in the executor.

## BEFORE - pollSize=100
- 10 tx/s: 150ms
- 25 tx/s: 200ms
- 50 tx/s: 300ms
- 75 tx/s: 5.2s
- 100 tx/s: 15s

## AFTER - pollSize=50
- 10 tx/s: 150ms
- 25 tx/s: 200ms
- 50 tx/s: 300ms
- 75 tx/s: 4.8s
- 100 tx/s: 14s

elevatebart committed:
fix: redirect to edit when saving new flow (#8560)

Co-authored-by: Piyush Bhaskar <102078527+Piyush-r-bhaskar@users.noreply.github.com>

Commit:
Merge branch 'develop' into feat/add_main_tenant

# Conflicts:
#	core/src/main/java/io/kestra/core/repositories/LocalFlowRepositoryLoader.java
#	core/src/test/resources/flows/valids/sleep-short.yml
#	webserver/src/test/java/io/kestra/webserver/controllers/api/ExecutionControllerRunnerTest.java

Commit:
wip(core): fix unit tests

Commit:
test(core): fix unit tests

Commit:
Merge branch 'develop' into feat/add_main_tenant

Commit:
test(core): fix unit tests

Commit:
test(core): fix unit tests

Skraye commented on cli/src/main/java/io/kestra/cli/AbstractApiCommand.java:
```diff
@@ -90,7 +90,7 @@ protected String apiUri(String path) {
         }
 
-        return tenantId == null ? "/api/v1" + path : "/api/v1/" + tenantId + path;
+        return tenantId == null ? "/api/v1/main" + path : "/api/v1/" + tenantId + path;
```
Use the MAIN_TENANT var from the TenantService

```suggestion
        return tenantId == null ? "/api/v1/main -> MAIN_TENANT" + path : "/api/v1/" + tenantId + path;
```

Commit:
feat(core): make tenant id required everywhere

Commit:
Merge branch 'develop' into feat/add_main_tenant

nkwiatkowski committed:
Merge branch 'develop' into feat/add_main_tenant

Commit:
feat(core): make tenant required in create user command

Commit:
Merge branch 'develop' into feat/add_main_tenant

Commit:
feat(core): clean the PR

Commit:
feat(core): add tenant id to dashboard controller

Commit:
Merge branch 'develop' into feat/add_main_tenant

# Conflicts:
#	core/src/test/java/io/kestra/core/repositories/AbstractFlowRepositoryTest.java

Commit:
fix(core): tests after merging

Commit:
Merge branch 'develop' into feat/add_main_tenant

# Conflicts:
#	core/src/test/java/io/kestra/core/runners/ExecutionServiceTest.java
#	webserver/src/test/java/io/kestra/webserver/controllers/api/ExecutionControllerRunnerTest.java

Commit:
Merge branch 'develop' into feat/add_main_tenant

loicmathieu commented on core/src/main/java/io/kestra/core/models/flows/FlowWithPath.java:
```diff
@@ -27,6 +27,7 @@ public class FlowWithPath {
 
     public static FlowWithPath of(FlowInterface flow, String path) {
         return FlowWithPath.builder()
+            .tenantId(flow.getTenantId())
```
This looks like a bug :( 
I wonder if it didn't need to be backported

loicmathieu commented on tests/src/main/java/io/kestra/core/utils/TestsUtils.java:
```diff
@@ -138,15 +136,12 @@ private static Execution mockExecution(StackTraceElement caller,
             .id(IdUtils.create())
             .executionId(execution.getId())
             .namespace(execution.getNamespace())
+            .tenantId(MAIN_TENANT)
```
Why not execution.getTenantId()?

loicmathieu commented on webserver/src/main/java/io/kestra/webserver/controllers/api/TemplateController.java:
```diff
@@ -88,6 +90,7 @@ public PagedResults<Template> find(
     public HttpResponse<Template> create(
         @Parameter(description = "The template") @Valid @Body Template template
     ) throws ConstraintViolationException {
+        template.setTenantId(MAIN_TENANT);
```
Should be (please apply everywhere)
```suggestion
        template.setTenantId(tenantService.resolveTenant());
```

loicmathieu commented on webserver/src/main/java/io/kestra/webserver/rooting/TenantAliasingRooter.java:
```diff
@@ -0,0 +1,35 @@
+import lombok.SneakyThrows;
+
+@Singleton
+@Requires(missingClasses = "io.kestra.ee.webserver.rooting.DefaultTenantAliasingRooter")
```
I'd rather not add that here, it shows that we have an override in EE which we try to never show (information disclosure vulnerability)

loicmathieu reviewed (changes requested):
Also please squash your commits.

If you need to backport the bug I notice (or think I notice), you can keep it in an additional commit.

I didn't check tests but if they pass, it should be good ;)

Commit:
clean(core): fixes after review

Commit:
Merge branch 'develop' into feat/add_main_tenant

loicmathieu reviewed (approved):
LGTM

Commit:
Merge branch 'develop' into feat/add_main_tenant

# Conflicts:
#	core/src/test/java/io/kestra/core/repositories/AbstractExecutionRepositoryTest.java
